### PR TITLE
Updated top bar implementations

### LINKED ISSUE #32: Place navigation burger in the top bar if the user is within a roadie
**User Story**
As a user, I want to navigate to different roadie planning pages through the top bar, so that I can view the details of my trip.

**Acceptance Criteria**

- Burger button shows in the top right corner of the top bar component, but only when the URL is within a specific roadie
- The burger button does not show up on the dashboard

**Additional Context**
N/A

**Dependencies**
N/A

### LINKED ISSUE #33: Road trip pages topbar
**User Story**
As a user, I want to always see a top bar that applies to the roadtrip I am in when I open a roadtrip.

**Acceptance Criteria**
* A TopBar component (perhaps we have two - HomeTopBar and RoadTripTopBar) that shows on all road trip pages
* For now - the top bar shows the name of the roadtrip we are in and has dummy add user and user profile icons (as per design)
* A modal that pops up when the add user button is clicked to allow the option to add attendees to the roadtrip
* Modal also shows the list of attendees (use dummy data for now)

**Additional Context**
image

### LINKED ISSUE #37: Change styling of topbar to match colour scheme
**User Story**
As a user, I want the colouring to be consistent across the web application, so that it is nice to look at.

**Acceptance Criteria**

- Make heading, buttons, and icon colouring and sizing match the colour scheme and other roadie page designs

**Additional Context**
N/A

**Dependencies**
N/A

### PR DESCRIPTION
**Pull request by lhuijser**
Changes:

- Matched styling of top bars to the colour scheme
- Split top bar into two - one for the dashboard, one for within specific road trips
- Added attendee list modal for the road trip top bar (ability to add new attendee to roadie only shows if the user is an organiser)
- Return to dashboard from within a roadie by clicking on home icon
- Road trip top bar title now equals the name of the roadie the user is in

Images:
Roadie specific top bar:
Screen Shot 2021-05-02 at 2 12 56 PM
Dashboard top bar:
Screen Shot 2021-05-02 at 2 13 03 PM
Attendee modal:
Screen Shot 2021-05-01 at 11 43 50 AM

Closes #33 
Closes #37 
Closes #32

### CODE REVIEW

File: frontend/src/components/HomeTopBar/HomeTopBar.jsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { Button, AppBar, Toolbar, withStyles, makeStyles } from '@material-ui/core';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import styles from './HomeTopBar.module.css';
+// import * as PropTypes from "prop-types";
+
+// TODO Flags for host
```
Comment (Denijar) [inline frontend/src/components/HomeTopBar/HomeTopBar.jsx:7]: hmm not sure why this commenting is still here, this was removed in a PR from a while ago. Maybe it was still in your local and it stayed there through the merge? neway pls delete <3

File: frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import GroupAddRoundedIcon from '@material-ui/icons/GroupAddRounded';
+import HomeRoundedIcon from '@material-ui/icons/HomeRounded';
+import { NavLink, useParams } from 'react-router-dom';
+import SideBar from '../SideBar/SideBar';
+import styles from './RoadTripTopBar.module.css';
+import useGet from '../../hooks/useGet';
+import AttendeesModal from './AttendeesModal/AttendeesModal';
+// import * as PropTypes from "prop-types";
+
+// TODO Flags for host
```
Comment (Denijar) [inline frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx:13]: yeah same here

File: frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+  };
+
+  return (
+    <CustomTopBar position="static">
+      <Toolbar variant="dense">
+        <SideBar />
+        <p className={styles.title}>{response?.data.name}</p>
+        <div className={classes.grow} />
+        <IconButton>
+          <GroupAddRoundedIcon onClick={handleOpenModal} />
```
Comment (Denijar) [inline frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx:67]: is it possible to have this icon be one without a "+" when you are not an organiser? e.g. `import GroupIcon from '@material-ui/icons/Group';` instead

**Review by Denijar — CHANGES_REQUESTED** [review]
Looks sick - im super excited to link up the adding attendees. Feels like everything is coming together.

image
Currently this modal doesn't match the style of the other modals we have been using in the app. e.g compared to (and also having the "cancel" on the right vs the left)
image

Personally I like the style you have done for this PR, but it would be good if we picked a style and were consistent because I have been implementing other modals using the Map page style.

File: frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx (hunk 3 above)
Comment (lhuijser) [reply frontend/src/components/RoadTripTopBar/RoadTripTopBar.jsx:67]: Ahh good catch, I'll include this now

**Review by Jennak21 — CHANGES_REQUESTED** [review]
Looks amazing Flynn! Got a couple of suggestions.  The AddButton component has just been merged in, so would be good to use that. Also, I think the withStyles code for a component should be in a separate file (e.g create a file called HomeTopBar.styles.js) alike to what has been done in the rest of the project. This is so that styling code remains separate from the rest of the code.

File: frontend/src/components/HomeTopBar/HomeTopBar.jsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { Button, AppBar, Toolbar, withStyles, makeStyles } from '@material-ui/core';
+import AccountCircleIcon from '@material-ui/icons/AccountCircle';
+import styles from './HomeTopBar.module.css';
+
+const CustomButton = withStyles({
```
Comment (Jennak21) [inline frontend/src/components/HomeTopBar/HomeTopBar.jsx:6]: Would be good to use the reusable AddButton component

**Review by courtneyeh — APPROVED** [review]
Looks great! Functional and pretty :)

**Review by Denijar — APPROVED** [review]
ya looks sick :D

**Pull request merged**